packages/mobile/src/escrow/saga.ts

```diff
@@ -117,6 +222,167 @@ function* registerStandbyTransaction(context: TransactionContext, value: string,
+      throw Error('Couldnt find own phone hash or pepper. Should never happen.')
+    }
+
+    const { phoneHash, pepper } = phoneHashDetails
+    const [contractKit, walletAddress, mtwAddress, feelessVerificationState]: [
+      ContractKit,
+      string,
+      string,
+      FeelessVerificationState
+    ] = yield all([
+      call(getContractKit),
+      call(getConnectedUnlockedAccount),
+      select(mtwAddressSelector),
+      select(feelessVerificationStateSelector),
+    ])
+
+    const komenciKit: KomenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    const [stableTokenWrapper, escrowWrapper, mtwWrapper]: [
+      StableTokenWrapper,
+      EscrowWrapper,
+      MetaTransactionWalletWrapper
+    ] = yield all([
+      call([contractKit.contracts, contractKit.contracts.getStableToken]),
+      call([contractKit.contracts, contractKit.contracts.getEscrow]),
+      call([contractKit.contracts, contractKit.contracts.getMetaTransactionWallet], mtwAddress),
+    ])
+
+    const escrowPaymentIds: string[] = yield call(
+      [escrowWrapper, escrowWrapper.getReceivedPaymentIds],
+      phoneHash
+    )
+
+    console.log('Payment IDs: ', escrowPaymentIds)
+
+    if (escrowPaymentIds.length === 0) {
+      Logger.debug(TAG + '@withdrawFromEscrow', 'No pending payments in escrow')
+      ValoraAnalytics.track(OnboardingEvents.escrow_redeem_complete)
+      return
+    }
+
+    const paymentIdSet: Set<string> = new Set()
+    for (const paymentId of escrowPaymentIds) {
+      paymentIdSet.add(paymentId)
+    }
+
+    const context = newTransactionContext(TAG, 'Withdraw from escrow')
+    // TODO: Batch the tranasctions and submit them together via `executeTransactions`
+    // method on an instance of the MTW then submitting like usual
+    const withdrawTxSuccess: boolean[] = []
+    // Using an upper bound of 1000 to be sure this doesn't run forever
+    for (let i = 0; i < 1000 && paymentIdSet.size > 0; i += 1) {
+      const { paymentId, privateKey } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
+      if (!paymentIdSet.has(paymentId)) {
+        continue
+      }
+      paymentIdSet.delete(paymentId)
+
+      const receivedPayment = yield call(getEscrowedPayment, escrowWrapper, paymentId)
+      const value = new BigNumber(receivedPayment[3])
+      if (!value.isGreaterThan(0)) {
+        Logger.warn(TAG + '@withdrawFromEscrowUsingPepper', 'Escrow payment is empty, skipping.')
+        continue
+      }
+
+      const withdrawTx: CeloTransactionObject<boolean> = yield formEscrowWithdrawTxWithNoCode(
+        contractKit,
+        mtwWrapper,
+        escrowWrapper,
+        stableTokenWrapper,
+        paymentId,
+        privateKey,
+        walletAddress,
+        mtwAddress,
+        value
+      )
+
+      try {
+        if (!komenciActive) {
+          yield call(sendTransaction, withdrawTx.txo, walletAddress, context)
```

Comment:
IIUC this will result a call to `Escrow.withdraw` from the EOA, but it is only the MTW that has permission to withdraw, so this will fail.

Reply:
I think this is already being done `formEscrowWithdrawTxWithNoCode`

Reply:
For posterity @tarikbellamine is right, this would work b/c we are batching the txs via MTW